lib/webapp.js

```diff
@@ -1001,9 +1001,19 @@ WebServiceHandle.prototype = {
             
           }
 
+          // Populating req.query with the query parameters
+          const queryString = path.includes('?') ? path.substring(path.indexOf('?') + 1) : '';
+          if(queryString !== '') {
+            const queryStringArray = queryString.split('&');
+            queryStringArray.forEach(pair => {
+              pair = pair.split('=');
+              req.query[pair[0]] = pair[1] || '';
+            })
+          }
+
           if (options.body) {
-            req._body = options.body;
-            req.body = options.body;
+            req._body = JSON.parse(options.body);
+            req.body = JSON.parse(options.body);
```

Comment:
This has several problems.
1) if you parse and its not json, you will throw an exception
2) options.body type detection is already done around line 913
3) line 923 does JSON.stringify(), so this just reverses it. Choose one or the other.
4) isn't stringify better since thats what the network does? what issue is this solving?

Reply:
Yes, stringify is better.
But Robert's team had created the test plugins to reproduce few errors.
And one of the errors occurred because they were accepting the parsed JSON body.

And hence I used JSON.parse() only for the requests that are routed internally. 
So we still need to do JSON.stringify() at line 923 for other requests.

Reply:
Thank you.
Please let me know if the changes looks good to you or it's still more safer to use the try catch instead of if else.